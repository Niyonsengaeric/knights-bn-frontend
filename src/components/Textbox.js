import React from 'react';
import '../assets/styles/containers/landingPage.scss';
import { Form } from 'react-bootstrap';

const Textbox = (props) => (
  <Form.Group>
    <Form.Control
      type={props.type}
      placeholder={props.placeholder}
      id={props.id}
      name={props.name}
      onChange={(event) => props.onChange(event.target)}
      className={props.validation}
    />
    <Form.Control.Feedback type="invalid">
      {props.errorMsg}
    </Form.Control.Feedback>
  </Form.Group>
);
export default Textbox;
