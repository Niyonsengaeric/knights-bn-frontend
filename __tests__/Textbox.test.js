import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import ReactTestRenderer from 'react-test-renderer';
import Textbox from '../src/components/Textbox';

describe('Test the Textbox ', () => {
  afterEach(cleanup);
  it('renders without crushing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Textbox />, div);
  });
  it('matches snapshot', () => {
    const tree = ReactTestRenderer.create(
      <Textbox id="button" placeholder="email" />
    );
    expect(tree).toMatchSnapshot();
  });
});
