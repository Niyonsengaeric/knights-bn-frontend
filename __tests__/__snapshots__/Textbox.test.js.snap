// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Textbox  matches snapshot 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control"
    id="button"
    onChange={[Function]}
    placeholder="email"
  />
  <div
    className="invalid-feedback"
  />
</div>
`;
